impl/src/main/java/org/apache/myfaces/push/Json.java

```diff
@@ -111,12 +112,45 @@ else if (object instanceof Class<?>)
         {
             encode(((Class<?>) object).getName(), builder);
         }
+        else if (object instanceof Record)
+        {
+            encodeRecord(object, builder);
+        }
         else
         {
             encodeBean(object, builder);
         }
     }
 
+    private static void encodeRecord(Object recordObject, StringBuilder builder) {
+
+        builder.append('{');
+        boolean fieldsFound = false;
+
+        for (Field field : recordObject.getClass().getDeclaredFields())
+        {
+            field.setAccessible(true);
```

Comment:
I could be wrong but is this the kind of thing they are removing in JDK 21? Being able to change field access?  I don't follow those JDK reflection updates much but i thought they were removing a lot of things?